### DISCUSSION TITLE
Emcee Sampler example notebook (emcee.ipynb) gives errors after upgrading Phoebe to 2.3.53

### DISCUSSION THREAD

**Discussion by janeyapyx** [phoebe-project/phoebe2#526] [Questions]
Hi, I wonder if I am the only one that faces issues with plotting and run.compute() especially in emcee. 
I am on Phoebe 2.3.53, astropy 4.3.1, Matplotlib 3.4.3 
and I have been facing problems in generating model light curves 
e.g.
b.run_compute(sample_from='ball_around_guess', 
              sample_num=20, model='init_from_model')

ValueError: single model exceeded number of allowed retries.  Check sampling distribution to ensure physical models exist.  Latest model error: failed to pass checks
Run Checks Report: FAIL
ERROR: constraints ['XXXXX'] failed to run.  Address errors and try again.  Call run_failed_constraints to see the tracebacks. (1 affected parameters, affecting system,run_compute)

'XXXXX' being one of the system parameters. 

While b.run_solver() seems to work fine, print(b.adopt_solution(trial_run=True)) gives error:
TypeError: cannot divide MVSamplesSlice by type <class 'numpy.int64'>
as well as in b.adopt_solution().

I also have plotting issues as well with the plot extends outside of the subplot's dimension. (Issue only occurs with figure that has more than 1 plot such as in  
_ = b.plot(x='phases', y='residuals', show=True) or 
_ = b.plot(s=0.005,solution='emcee_sol', style='trace', show=True) ).

If no one else has the same problem then perhaps I am missing some dependencies or working with the wrong versions.

Thanks.

**Comment by kecnry** [comment]
It looks like there were more issues introduced by changes to astropy (I'm guessing your installation of astropy updated alongside phoebe).  

We'll continue working to track these bugs down and fix them, but in the meantime you should be able to avoid them by downgrading astropy slightly (I'm not sure the exact version - but please do report back if you can track down when they were introduced).

**Reply by kecnry** [reply]
This should be fixed in the [2.3.54 bugfix release](https://github.com/phoebe-project/phoebe2/releases/tag/2.3.54).  If anyone else runs into a similar bug on the latest version, please [report the issue](https://github.com/phoebe-project/phoebe2/issues/new) and include a script to reproduce.  The emcee notebook in the docs has been confirmed to run successfully on 2.3.54.

**Reply by janeyapyx** [reply]
Thanks a lot for the quick response! (and sorry I could't get back to you sooner) When I upgrade my existing environment to 2.3.54, my problems still persist and then I realize that might be due to the version of python I was on (3.9), so I downgraded it and now I can run the emcee notebook without much problem. That being said, the combination of package versions that works for me currently are:
python 3.7.11 (or 3.7.10)
astropy 4.3.1
Matplotlib 3.4.2 (I have plotting errors with 3.4.3)
Phoebe 2.3.54

And in case you might be interested, I also tried running on python3.8 and everything seems to work perfect except for run.compute() during parameters sampling, same kind of error I reported above, i.e. ERROR: constraints ['XXXXX'] failed to run, and code exited.

Anyway, I really appreciate your help and I am happy with the current combination, so no further questions from me! Thanks.

**Reply by kecnry** [reply]
Ok, I'm glad you have a working setup now, but I'd still like to fix any remaining issues with other versions.  Can you tell me the details of your 3.9 setup and whether it was causing the same error (the `_ = b.plot_distribution_collection(distribution='ball_around_guess', parameters=['asini@binary'])` line) or somewhere else?

And what plotting command/line gives you plotting errors with matplotlib 3.4.3?

Thanks!
Kyle

**Reply by janeyapyx** [reply]
Sure, but since I am running the script with the current setup, so I created a separate conda environment to reproduce the errors I had.
I paste below screenshots with errors running on Python3.9 (run.compute() ) and matplotlib3.4.3 (plot extends outside of box), in chronological order following emcee.ipynb notebook.

<img width="2085" alt="Screen Shot 2021-09-20 at 2 39 24 PM" src="https://user-images.githubusercontent.com/55396806/133964398-0f5b436b-84aa-4451-9002-fa31ffe4bff0.png">

<img width="598" alt="Screen Shot 2021-09-20 at 2 42 36 PM" src="https://user-images.githubusercontent.com/55396806/133964627-992add2f-0604-4568-9f79-5de462f3f83d.png">

In case it might be of help, you can find the list of packages I use in this pdf:
[ex.pdf](https://github.com/phoebe-project/phoebe2/files/7194021/ex.pdf)
